### ISSUE TITLE
GATConv code questions

### ISSUE THREAD

**Issue by quan-jiayi** [dmlc/dgl#1278]
## ❓ Questions and Help

Hello,will you give me a help?
about the GATConv source code in [GATConv](https://docs.dgl.ai/_modules/dgl/nn/pytorch/conv/gatconv.html#GATConv)

`feat = self.fc(h).view(-1, self._num_heads, self._out_feats)
        el = (feat * self.attn_l).sum(dim=-1).unsqueeze(-1)
        er = (feat * self.attn_r).sum(dim=-1).unsqueeze(-1)
        graph.ndata.update({'ft': feat, 'el': el, 'er': er})
        # compute edge attention
        graph.apply_edges(fn.u_add_v('el', 'er', 'e'))
        e = self.leaky_relu(graph.edata.pop('e'))
        # compute softmax
        graph.edata['a'] = self.attn_drop(edge_softmax(graph, e))
        # message passing
        graph.update_all(fn.u_mul_e('ft', 'a', 'm'),
                         fn.sum('m', 'ft'))`

here:
(1) what is the meaning of el and er ?
(2) what is the meaning of fn.u_add_v,fn.u_mul_e ,I have found them in https://docs.dgl.ai/api/python/function.html ,however the introductions are so brief and there is no any examples。
(3) why there is no concatenate operation is the source code，I think this code is very different from the paper。

thank you!

**Comment by yzh119** [comment]
I closed this issue because it's a duplicate of #1279 .
